examples/src/main/scala/zio/config/examples/ReadWriteReport.scala

```diff
@@ -2,91 +2,89 @@ package zio.config.examples
 
 import zio.DefaultRuntime
 import zio.config._, Config._
-import zio.config.actions.ConfigDocs
-import zio.config.actions.ConfigDocs.KeyDescription
+import zio.config.actions.ConfigDocs._
 
 object ReadWriteReport extends App {
 
   case class Password(value: String)
   case class UserPwd(name: String, pwd: Option[Password], abc: Option[String], value: Option[XYZ])
   case class Token(value: String, clientid: String)
-  case class XYZ(xyz: String, someInteger: Int)
+  case class XYZ(xyz: String, someInteger: Either[Int, String])
 
   type ProdConfig = Either[UserPwd, Token]
 
   // An example where user provides a description once and for all, and use it for read, write, report!
   val config: ConfigDescriptor[ProdConfig] =
-    ((string("usr") ~ "Example: some-user" |@|
-      string("pwd").xmap(Password)(_.value).optional ~ "We don't care" ~ "yea !" |@|
-      string("jhi").optional ~ "Example: ghi" |@|
-      (string("xyz") |@| int("abc"))(XYZ.apply, XYZ.unapply).optional ~ "Example: xyz")(
+    ((string("usr") ? "Example: some-user" |@|
+      string("pwd").xmap(Password)(_.value).optional ? "sec" |@|
+      string("jhi").optional ? "Ex: ghi" |@|
+      (string("xyz") |@| int("abc").orElseEither(string("def")))(XYZ.apply, XYZ.unapply).optional ? "Ex: ha")(
       UserPwd.apply,
       UserPwd.unapply
-    ) or
-      (string("auth_token") |@| string("clientid"))(Token.apply, Token.unapply)) ~ "Prod Config"
+    ) orElseEither
```

Comment:
I think we should introduce `<+>` as the alias for `orElseEither` (it being the dual of `<*>` / `zip`).